src/server/utils/stateConfig.js

```diff
@@ -123,8 +123,7 @@ export const setStateConfig = (providedStateConfig) => {
             throw makeConfigEnvError();
           }
           acc.client[key] = client[configEnv];
-        }
-        if (typeof client === 'string') {
+        } else {
```

Comment:
keep validating the type

Reply:
So we want to only explicitly accept string, number, or boolean? The reason I did this was it puts the complexity to 13. I talked with @JAdshead and we came up with this solution.